### DISCUSSION TITLE
Comparing eia861 to allocation of eia923 to eia860 generator-owners

### DISCUSSION THREAD

**Discussion by aesharpe** [catalyst-cooperative/pudl#2406] [Ideas]
From @jrea-rmi:

Maybe @katherinelamb and @TrentonBush have thoughts

> I said I'd let you know what I found when comparing eia861 to our allocation of eia923 to eia860 generator-owners, here it is!
> Out of 70 subsidiaries that have both FERC & IRP data (so this difference creates problems for IRP projections that feed into EnAct),
> 
> - 56 have net generation within 10% from both sources
> - 5 have higher net generation in operations_emissions
> - 9 have lower net generation in operations_emissions
> 
> The biggest problems for combination % error and size of company are: Wisconsin Electric Power, Evergy Metro, Evergy Kansas South, Gulf Power.
> 
> I looked into a few utilities that don't match very well, and found that it's common for them to have a significant fraction of their generation from multi-owned plants. For example, Empire District Co has 5.6 TWh of owned generation in our allocation of eia923 data, but 4.0 TWh of owned generation in eia861. Three big plants make up 40% of Empire District Co's generation, and Empire District Co owns 17% of those plants' total generation according to our eia923 allocation.
> 
> It feels likely that this problem is because the ownership fractions in eia860 aren't the same as the fraction of generation that each owner gets from the plant. It's possible that the allocation process in PUDL could include reconciliation with eia861, but that sounds quite difficult and would require some **iteration or optimization routine to minimize the total net generation error, finding the right fractions of generation to give to each owner by modifying a generation ownership fraction**. That might not be worth the effort now, but at least I've pursued a couple hunches and have reasonable confidence in what the problem is.

![microsoftteams-image__2_](https://user-images.githubusercontent.com/49878195/225383267-c4d767ce-44a1-445b-bc07-e6d40ac5f89a.png)

**Comment by TrentonBush** [comment]
This is an interesting comparison and good error analysis! As far as making a solver to figure out allocation fraction, I agree that that is probably too much work for this one problem, but the good/bad news is that I think we have a few related problems that could all benefit from that kind of tooling. Maybe it's worth it if the cost is shared between them?

The EIA923 fuel price imputation has a similar allocation problem where half the prices are missing but we have many different aggregates involving those prices, and in principle we can estimate the redacted values using those.

Also, just getting a good graph data structure and framework I think would help with all these graph-style problems: boiler-generatior association, generator-owner, generator-fuel, EPA/EIA unit relationships, etc
